tracer/test/test-applications/integrations/Samples.VersionConflict.2x/Samples.VersionConflict.2x.csproj

```diff
@@ -0,0 +1,9 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  
+  <ItemGroup>
+    <PackageReference Include="Datadog.Trace" Version="2.255.251" />
```

Comment:
Once we release v2.0.0, should we change this to be 2.0.0, to give a more realistic take on the version conflict. My concern is that this package will be a "newer version with older code", which isn't a scenario we realistically need to test. 

That said, do we need tests for the version conflict for both cases?
* Older (post 2.0.0) manual, newer automatic
* Older (post 2.0.0) automatic, newer manual

Or do we just rely on the compatibility tests we're going to implement down the line?

Reply:
We definitely need both kinds of test (newer + older and older + newer), if only because .NET Core has special handling for the older + newer scenario.
I think the "2.255.251" will remain as-is, and I need to add a test for an older version (in the future it will be 2.0, but right now it's a bit tricky to write since we haven't updated the version number yet :/ )

Reply:
> I think the "2.255.251" will remain as-is

My concern is along the lines of: 
* we add a new method/format in 2.1, which is backwards compatible with 2.0
* the native loader sees the manual package is 2.255.251, so assumes it has that change
* 💥 

Obviously ways around that etc, just wondering if we need to/should keep 2.255.251 up to date as the "latest" release. That's a PITA if so, and something we can punt on till this test fails I guess!

Reply:
That's a fair concern, though I doubt we will implement version-based compatibility rules (I believe it would be more along the lines of "does this version of Datadog.Trace exposes a `IAutomaticTracer2` interface?")

But since this concern would be costly to address, so let's cross this bridge when we come to it 🙂